daemon/cluster/noderunner.go

```diff
@@ -271,7 +271,13 @@ func (n *nodeRunner) enableReconnectWatcher() {
 		if n.stopping {
 			return
 		}
-		config.RemoteAddr = n.cluster.getRemoteAddress()
+		remotes := n.cluster.getRemoteAddressList()
+		if len(remotes) > 0 {
+			config.RemoteAddr = remotes[0]
+		} else {
+			config.RemoteAddr = ""
+		}
```

Comment:
I'm trying to understand what this code is for. Shouldn't the remote address (if any) have been filled in when the nodeRunner was created, and persisted on disk? Why is it being overwritten here? cc @tonistiigi